all/core/state.py

```diff
@@ -387,6 +387,49 @@ def shape(self):
     def __len__(self):
         return self.shape[0]
 
+    @classmethod
+    def cat(cls, state_array_list, axis=0):
+        '''Concatenates along batch dimention'''
+        if len(state_array_list) == 0:
+            raise ValueError("cat accepts a non-zero size list of StateArrays")
+
+        d = {}
+        state_size = sum(state_array.shape[axis] for state_array in state_array_list)
+        new_shape = list(state_array_list[0].shape)
+        new_shape[axis] = state_size
+        new_shape = tuple(new_shape)
+        keys = list(state_array_list[0].keys())
+        for key in keys:
+            d[key] = torch.cat([state_array[key] for state_array in state_array_list], axis=axis)
+        return StateArray(d, new_shape, device=state_array_list[0].device)
+
+    def flatten_for_execution(self, fn):
```

Comment:
This function seems a bit contrived.  Would it be clearer to just add a `flatten` method and let users handle resizing for `batch_execute`?  

```
target_values = states.flatten().batch_execute(256, compute_target_values).view(states.shape)
```

Reply:
Yeah, that is probably a better approach.

Reply:
Some weird things happening here in the unit tests I can't quite figure out.